src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -41,36 +49,29 @@ export class HumanDescriptionMapper {
 
     if (!sigHash) return null;
 
-    let token: Token | null = null;
-    try {
-      token = await this.tokenRepository.getToken({
-        chainId,
-        address: transaction.to,
-      });
-    } catch (error) {
-      this.loggingService.debug(`Error trying to get token: ${error.message}`);
-    }
-
     try {
-      const descriptionFragments =
+      const humanDescriptionFragments =
         this.humanDescriptionRepository.getHumanDescription({
           functionSignatureHash: sigHash,
           to: transaction.to,
           data: transaction.data,
         });
 
-      const description = this.createHumanDescription(
-        descriptionFragments,
-        token,
+      const richDecodedInfoFragments = await this.enrichFragments(
+        humanDescriptionFragments,
+        transaction,
+        chainId,
       );
 
-      const safeAppInfo = isMultisigTransaction(transaction)
-        ? await this.safeAppInfoMapper.mapSafeAppInfo(chainId, transaction)
-        : null;
+      const richDecodedInfo = await this.enrichSafeAppInfo(
+        richDecodedInfoFragments,
+        transaction,
+        chainId,
+      );
 
-      return safeAppInfo
-        ? `${description} via ${safeAppInfo.name}`
-        : description;
+      return {
+        fragments: richDecodedInfo,
+      };
```

Comment:
Nitpick: since we have a class for this entity in the route I would return the respective instance. Wdyt?

```suggestion
      return new RichDecodedInfo(richDecodedInfo);
```

Reply:
Nice find! Makes sense